apps/studio/src/store/modules/PinConnectionModule.ts

```diff
@@ -0,0 +1,77 @@
+import { PinnedConnection } from "@/common/appdb/models/PinnedConnection";
+import { SavedConnection } from "@/common/appdb/models/saved_connection";
+import { Module } from "vuex";
+import { State as RootState } from '../index';
+
+interface State {
+  pins: PinnedConnection[]
+}
+
+export const PinConnectionModule: Module<State, RootState> = {
+  namespaced: true,
+  state: () => ({
+    pins: []
+  }),
+  getters: {
+    pinned(state: State): PinnedConnection[] {
+      return state.pins;
+    },
+    orderedPins(_state, getters, rootState): PinnedConnection[] {
+      const connections = rootState['data/connections'].items;
+      return getters.pinned.sort((a, b) => a.position - b.position).map((pin: PinnedConnection) => {
+        const c = connections.find((c) => c.id === pin.connectionId);
+        if (c) pin.connection = c;
+        return c ? pin : null
+      }).filter((p) => !!p);
+    },
+    pinnedConnections(_state: State, getters): SavedConnection[] {
+      return getters.orderedPins.map((pin) => pin.connection);
+    }
+  },
+  mutations: {
+    set(state, pins: PinnedConnection[]) {
+      state.pins = pins;
+    },
+    add(state, newPin: PinnedConnection) {
+      state.pins.push(newPin);
+    },
+    remove(state, pin: PinnedConnection) {
+      state.pins = _.without(state.pins, pin);
+    }
+  },
+  actions: {
+    async loadPins(context) {
+      const pins = await PinnedConnection.find({
+        where: {
+          workspaceId: context.rootState.workspaceId
+        }
+      });
+      context.commit('set', pins || []);
+    },
+    async unloadPins(context) {
+      context.commit('set', []);
+    },
+    async add(context, item: SavedConnection) {
+      const existing = context.state.pins.find((p) => p.connectionId === item.id)
+      if (existing) {
+        return;
+      }
+
+      const newPin = new PinnedConnection(item);
+      newPin.position = (context.getters.orderedPins.reverse()[0]?.position || 0) + 1;
+      await newPin.save();
+      context.commit('add', newPin);
+    },
+    async reorder(_context, pins: PinnedConnection[]) {
```

Comment:
I see we do something similar in the PinModule. One difference is that you're not calling `set` here. You should be!